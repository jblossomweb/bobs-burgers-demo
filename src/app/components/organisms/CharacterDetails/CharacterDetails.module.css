.wrapper {
  @apply w-full flex flex-wrap sm:flex-nowrap flex-row justify-start items-start gap-8;
}

.card {
  width: 80vw;
  @apply sm:w-96;
}

.content {
  @apply w-full sm:w-1/2;
}

.content h1 {
  @apply text-4xl font-bold mb-4;
}

.content h2 {
  @apply text-2xl font-bold my-4;
}

.content p {
  @apply my-2 text-base;
}

.content ul {
  @apply my-2 text-base;
}

.avatar {
  @apply border-2 border-palette-dark-blue my-4;
}

.avatar:hover {
  @apply border-palette-light-blue;
}

.avatar:active {
  @apply translate-y-0.5;
}

.image {
  @apply w-full;
}

.related {
  @apply flex flex-row flex-wrap gap-2;
}

.links {
  @apply my-8;
}
