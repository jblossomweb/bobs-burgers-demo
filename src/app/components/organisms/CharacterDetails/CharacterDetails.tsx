import React from 'react'
import { Link } from 'react-router-dom'
import type Character from 'app/models/Character'
import Button from 'app/components/atoms/Button'
import Card from 'app/components/atoms/Card'
import CharacterAvatar from 'app/components/ions/CharacterAvatar'
import styles from './CharacterDetails.module.css'

export interface Props {
  character: Character
  relatives?: Character[]
}

const CharacterDetails: React.FC<Props> = ({ character, relatives }) => (
  <div className={styles.wrapper}>
    <Card
      className={styles.card}
      head={
        <img
          className={styles.image}
          src={character.image}
          alt={character.name}
        />
      }
    />
    <div className={styles.content}>
      <h1>{character.name}</h1>
      <p>{character.occupation}</p>
      <ul>
        <li>First Episode: {character.firstEpisode}</li>
        <li>Voiced By: {character.voicedBy}</li>
        <li>Gender: {character.gender}</li>
        {!!character.age && <li>Age: {character.age}</li>}
        {!!character.hairColor && <li>Hair: {character.hairColor}</li>}
      </ul>
      {!!relatives?.length && (
        <div>
          <h2>Related Characters:</h2>
          <div className={styles.related}>
            {relatives.map((relative) => (
              <CharacterAvatar key={relative.id} character={relative} />
            ))}
          </div>
        </div>
      )}
      <div className={styles.links}>
        <Link to="/characters">
          <Button hollow>&lt; All Characters</Button>
        </Link>
      </div>
    </div>
  </div>
)

export default CharacterDetails
