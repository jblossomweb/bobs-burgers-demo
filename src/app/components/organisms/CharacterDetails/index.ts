export { default, type Props } from './CharacterDetails'
