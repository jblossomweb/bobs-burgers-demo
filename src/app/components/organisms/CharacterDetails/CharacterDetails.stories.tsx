import React from 'react'
import type { ComponentStory, ComponentMeta } from '@storybook/react'
import mockCharacter from 'app/models/__mocks__/character.json'
import mockRelatives from 'app/models/__mocks__/relatives.json'

import CharacterDetails, { type Props } from './CharacterDetails'

export default {
  title: 'components/organisms/CharacterDetails',
  component: CharacterDetails,
  argTypes: {
    character: { control: 'object' },
    relatives: { control: 'object' },
  },
} as ComponentMeta<typeof CharacterDetails>

const Template: ComponentStory<typeof CharacterDetails> = ({
  character,
  relatives,
}: Props) => <CharacterDetails character={character} relatives={relatives} />

export const Sample = Template.bind({})
Sample.args = {
  character: mockCharacter,
}

export const Relatives = Template.bind({})
Relatives.args = {
  ...Sample.args,
  relatives: mockRelatives as Props['relatives'],
}
