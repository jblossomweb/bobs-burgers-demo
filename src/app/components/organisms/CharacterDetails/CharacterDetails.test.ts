import storySnaps from 'test/utils/storySnaps'
import * as stories from './CharacterDetails.stories'

describe('app/components/organisms/CharacterDetails', () => {
  describe('stories', () => {
    storySnaps(stories)
  })
})
