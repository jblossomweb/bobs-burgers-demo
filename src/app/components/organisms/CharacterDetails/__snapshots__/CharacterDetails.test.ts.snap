// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/components/organisms/CharacterDetails stories Relatives matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      class="container"
    >
      <div
        class="card card"
      >
        <img
          alt="Robert \\"Bob\\" Belcher, Jr."
          class="image"
          src="https://bobsburgers-api.herokuapp.com/images/characters/47.jpg"
        />
      </div>
    </div>
    <div
      class="content"
    >
      <h1>
        Robert "Bob" Belcher, Jr.
      </h1>
      <p>
        Chef/owner and operator of Bob's Burgers
      </p>
      <ul>
        <li>
          First Episode: "Human Flesh"
        </li>
        <li>
          Voiced By: H. Jon Benjamin
        </li>
        <li>
          Gender: Male
        </li>
        <li>
          Age: 46
        </li>
      </ul>
      <div>
        <h2>
          Related Characters:
        </h2>
        <div
          class="related"
        >
          <div
            class="avatar avatar"
            style="background-color: rgb(97, 218, 251);"
          >
            <div
              class="wrapper"
            >
              <a
                href="/characters/5"
              >
                <img
                  alt="Al"
                  class="image"
                  src="https://bobsburgers-api.herokuapp.com/images/characters/5.jpg"
                />
              </a>
            </div>
          </div>
          <div
            class="avatar avatar"
            style="background-color: rgb(97, 218, 251);"
          >
            <div
              class="wrapper"
            >
              <a
                href="/characters/45"
              >
                <img
                  alt="Robert \\"Big Bob\\" Belcher, Sr."
                  class="image"
                  src="https://bobsburgers-api.herokuapp.com/images/characters/45.jpg"
                />
              </a>
            </div>
          </div>
          <div
            class="avatar avatar"
            style="background-color: rgb(97, 218, 251);"
          >
            <div
              class="wrapper"
            >
              <a
                href="/characters/169"
              >
                <img
                  alt="Gerald G. Garrison"
                  class="image"
                  src="https://bobsburgers-api.herokuapp.com/images/characters/169.jpg"
                />
              </a>
            </div>
          </div>
          <div
            class="avatar avatar"
            style="background-color: rgb(97, 218, 251);"
          >
            <div
              class="wrapper"
            >
              <a
                href="/characters/170"
              >
                <img
                  alt="Gertie Humikowski"
                  class="image"
                  src="https://bobsburgers-api.herokuapp.com/images/characters/170.jpg"
                />
              </a>
            </div>
          </div>
          <div
            class="avatar avatar"
            style="background-color: rgb(97, 218, 251);"
          >
            <div
              class="wrapper"
            >
              <a
                href="/characters/180"
              >
                <img
                  alt="Goldie"
                  class="image"
                  src="https://bobsburgers-api.herokuapp.com/images/characters/180.jpg"
                />
              </a>
            </div>
          </div>
          <div
            class="avatar avatar"
            style="background-color: rgb(97, 218, 251);"
          >
            <div
              class="wrapper"
            >
              <a
                href="/characters/275"
              >
                <img
                  alt="Nurse Liz"
                  class="image"
                  src="https://bobsburgers-api.herokuapp.com/images/characters/275.jpg"
                />
              </a>
            </div>
          </div>
          <div
            class="avatar avatar"
            style="background-color: rgb(97, 218, 251);"
          >
            <div
              class="wrapper"
            >
              <a
                href="/characters/280"
              >
                <img
                  alt="Mall Santa"
                  class="image"
                  src="https://bobsburgers-api.herokuapp.com/images/characters/280.jpg"
                />
              </a>
            </div>
          </div>
          <div
            class="avatar avatar"
            style="background-color: rgb(97, 218, 251);"
          >
            <div
              class="wrapper"
            >
              <a
                href="/characters/483"
              >
                <img
                  alt="Trev"
                  class="image"
                  src="https://bobsburgers-api.herokuapp.com/images/characters/483.jpg"
                />
              </a>
            </div>
          </div>
          <div
            class="avatar avatar"
            style="background-color: rgb(97, 218, 251);"
          >
            <div
              class="wrapper"
            >
              <a
                href="/characters/500"
              >
                <img
                  alt="Zander"
                  class="image"
                  src="https://bobsburgers-api.herokuapp.com/images/characters/500.jpg"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="links"
      >
        <a
          href="/characters"
        >
          <button
            class="button hollow"
            type="button"
          >
            &lt; All Characters
          </button>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`app/components/organisms/CharacterDetails stories Sample matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      class="container"
    >
      <div
        class="card card"
      >
        <img
          alt="Robert \\"Bob\\" Belcher, Jr."
          class="image"
          src="https://bobsburgers-api.herokuapp.com/images/characters/47.jpg"
        />
      </div>
    </div>
    <div
      class="content"
    >
      <h1>
        Robert "Bob" Belcher, Jr.
      </h1>
      <p>
        Chef/owner and operator of Bob's Burgers
      </p>
      <ul>
        <li>
          First Episode: "Human Flesh"
        </li>
        <li>
          Voiced By: H. Jon Benjamin
        </li>
        <li>
          Gender: Male
        </li>
        <li>
          Age: 46
        </li>
      </ul>
      <div
        class="links"
      >
        <a
          href="/characters"
        >
          <button
            class="button hollow"
            type="button"
          >
            &lt; All Characters
          </button>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
